pkg/qemu/qemu.go

```diff
@@ -64,13 +64,13 @@ func (q *QEMU) CmdLine() []string {
 	// - earlyprintk=ttyS0: print very early debug messages to the serial
 	// - console=ttyS0: /dev/console points to /dev/ttyS0 (the serial port)
 	// - q.KernelArgs: extra, optional kernel arguments
-	args = append(args, "-append", "console=ttyS0 earlyprintk=ttyS0")
-	if q.KernelArgs != "" {
-		args[len(args)-1] += " " + q.KernelArgs
+	if q.Kernel != "" {
+		args = append(args, "-kernel", q.Kernel)
+		args = append(args, "-append", "console=ttyS0 earlyprintk=ttyS0")
+		if q.KernelArgs != "" {
```

Comment:
it might be worth checking to see if q.Kernel is "" and q.KernelArgs is not, because it means someone is confused at the command line :-)

Reply:
Since this function does not return error that would have to be changed (or panic()). I'll leave it as is as I'm not comfortable adding either.